src/Orchard.Cms.Web/Modules/Orchard.OpenId/Services/OpenIdConfiguration.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.IdentityModel.Tokens.Jwt;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OpenIddict;
+using Orchard.OpenId.Services;
+using Orchard.OpenId.Settings;
+
+namespace Orchard.OpenId
+{
+    public class OpenIdConfiguration : IConfigureOptions<OpenIddictOptions>
+    {
+        private readonly ILogger<OpenIdConfiguration> _logger;
+        private readonly IOpenIdService _openIdService;
+
+        public OpenIdConfiguration(ILogger<OpenIdConfiguration> logger, IOpenIdService openIdService)
+        {
+            _logger = logger;
+            _openIdService = openIdService;
+        }
+
+        public void Configure(OpenIddictOptions options)
+        {   
+            var settings = _openIdService.GetOpenIdSettingsAsync().Result;
+            if (!_openIdService.IsValidOpenIdSettings(settings))
```

Comment:
We throw later in the same method if there is a config error. We should here too, and log the error.